dbt/adapters/spark/connections.py

```diff
@@ -74,6 +74,7 @@ class SparkCredentials(Credentials):
     connect_timeout: int = 10
     use_ssl: bool = False
     server_side_parameters: Dict[str, Any] = field(default_factory=dict)
+    spark_conf_string: Optional[str] = None
```

Comment:
why does this need to be a string? Can this be a dict? Doing so would allow us to escape special characters in the configuration

Reply:
> Just to be clear, my purpose is to be able to change the Spark configuration with profile.yml when connecting via Thrift connection. These configs will be adjusted according to the DBT models.
> 
> To avoid change detection on profile.yml and do a full re-parse([docs](https://docs.getdbt.com/reference/parsing)) . The simplest way is to pass these configs from environment variables through the jinja format([ref](https://github.com/dbt-labs/dbt-core/issues/5600)). `spark_conf_string: "{{ env_var('SPARK_CONFIG_STRING') }}" `
> 
> With the server_side_parameters approach, its data type is dict, it is difficult to bypass the re-parse limitation if we want to use the environment variable mentioned above.
> 
> With these specific requirements, it led me to create another parameter with the String data type.

Hi @colin-rogers-dbt , 
As I mentioned in [the previous comment](https://github.com/dbt-labs/dbt-spark/pull/591#issuecomment-1432605791), the main reason is to prevent full-reparse when we need to change the spark configuration via dbt profile sequentially

Reply:
my bad, should have made the connection.